Repository/NpmRepository.php

```diff
@@ -147,10 +148,19 @@ protected function createArrayRepositoryConfig(array $packageConfigs)
         $loader = new ArrayLoader();
 
         foreach ($packageConfigs as $version => $config) {
-            $config['version'] = $version;
-            $config = $this->assetType->getPackageConverter()->convert($config);
-            $config = $this->assetRepositoryManager->solveResolutions($config);
-            $packages[] = $loader->load($config);
+            try {
+                $config['version'] = $version;
+                $config = $this->assetType->getPackageConverter()->convert($config);
+                $config = $this->assetRepositoryManager->solveResolutions($config);
+                $packages[] = $loader->load($config);
+            } catch (\UnexpectedValueException $exception) {
+                // Most probably version constraint is broken.
+                // Skip this version and hope that another one will be OK
+                $this->io->write("<warning>Skipped {$config['name']} version {$version}: {$exception->getMessage()}</warning>", IOInterface::VERBOSE);
+                continue;
+            } catch (\Exception $exception) {
+                throw $exception;
```

Comment:
why catch it if you don't do anything with it?

Reply:
@SilverFire See my previous review, because it is past in outdated status with your new update, but it's not the case: this catch is unnecessary.

Reply:
I also think this can be removed, you're just catching and throwing the exception again :)

Reply:
Opps. It used id for debug to place breakpoint on `throw $exception`, then I just forgot to remove it. Thank you for pointing out, friends :)